router/batchrouter/batchrouter.go

```diff
@@ -2297,8 +2286,10 @@ func (brt *HandleT) Setup(backendConfig backendconfig.BackendConfig, jobsDB, err
 		// error is ignored as context.TODO() is passed, err is not expected.
 		_ = brt.reporting.WaitForSetup(context.TODO(), types.CoreReportingClient)
 	}
-	if brt.warehouseURL == "" {
-		brt.warehouseURL = misc.GetWarehouseURL()
+	if brt.warehouseClient == nil {
+		brt.warehouseClient = client.NewWarehouse(misc.GetWarehouseURL(), client.WithTimeout(
+			config.GetDuration("WarehouseClient.timeout", 30, time.Second),
```

Comment:
```suggestion
			config.GetDuration("Warehouse.client.timeout", 30, time.Second),
```
Should it be something like this?

Reply:
Semantically, I felt that `Warehouse.client` -> means a client that runs in a warehouse.

Where `WarehouseClient` is the warehouse client in general.